cli/src/commands/fix.rs

```diff
@@ -270,16 +302,46 @@ fn fix_file_ids<'a>(
     tool_inputs.into_par_iter().try_for_each_init(
         || updates_tx.clone(),
         |updates_tx, tool_input| -> Result<(), BackendError> {
-            let mut read = store.read_file(&tool_input.repo_path, &tool_input.file_id)?;
-            let mut old_content = vec![];
-            read.read_to_end(&mut old_content).unwrap();
-            if let Ok(new_content) = run_tool(tool_command, tool_input, &old_content) {
-                if new_content != *old_content {
-                    let new_file_id =
-                        store.write_file(&tool_input.repo_path, &mut new_content.as_slice())?;
-                    updates_tx.send((tool_input, new_file_id)).unwrap();
+            // The first matching tool gets its input from the committed file, and any
+            // subsequent matching tool gets its input from the previous matching tool's
+            // output. In either case, we store the input in `prev_content`, but we avoid
+            // reading anything until we know at least one tool matches.
+            let mut prev_content: Option<Vec<u8>> = None;
+            let mut any_changes: bool = false;
+            for tool_config in tools_config.tools.iter() {
+                if tool_config.matcher.matches(&tool_input.repo_path) {
+                    if prev_content.is_none() {
+                        let mut content = vec![];
+                        let mut read =
+                            store.read_file(&tool_input.repo_path, &tool_input.file_id)?;
+                        read.read_to_end(&mut content).unwrap();
+                        prev_content = Some(content);
+                    }
```

Comment:
nit: perhaps, this `Option<_>` can be removed by peeking matching tools. Something like
```rust
let mut matched_tools = tools.iter().filter(|tool| tool.matcher.matches(..)).peekable();
if matched_tools.peek().is_none() {
    return;
}
read_file(..);
for tool in matched_tools {
...
```